File: 1196/b/b.cc
#include <cstdio>
#include <iostream>
#include <vector>

using namespace std;
typedef long long ll;

class Solution {
public:
  void solve() {
    int n, k;
    scanf("%d%d", &n, &k);
    ll curr = 0;
    vector<int> ans;
    ans.reserve(k);
    for (int i = 0; i < n; ++i) {
      int a;
      scanf("%d", &a);
      curr += (ll)a;
      if (curr % 2 == 1 && ans.size() < k) {
        ans.emplace_back(i + 1);
        curr = 0;
      }
    }
    if (ans.size() == k && curr % 2 == 0) {
      ans[k - 1] = n;
      printf("YES\n");
      for (int i : ans)
        printf("%d ", i);
      printf("\n");
    } else
      printf("NO\n");
  }
};

int main() {
  ios::sync_with_stdio(false);
  cin.tie(0);
  int t;
  scanf("%d", &t);
  while (t--) {
    Solution solution = Solution();
    solution.solve();
  }
}

File: 1196/b/b.kt
// Did not pass all tests because of slow I/O

fun readInt(): Int {
    return readLine()!!.toInt()
}

fun readInts(): List<Int> {
    return readLine()!!.split(" ").map(String::toInt)
}

class Solution(private val n: Int, private val k: Int, private val a: List<Int>) {
    fun solve() {
        var curr = 0L
        val ans = mutableListOf<Int>()
        for (right in 0 until n) {
            curr += a[right].toLong()
            if (curr % 2 == 1L && ans.size < k) {
                ans.add(right + 1)
                curr = 0
            }
        }
        if (ans.size == k && curr % 2 == 0L) {
            ans[k - 1] = n
            println("YES")
            println(ans.joinToString(" "))
        } else
            println("NO")

    }
}

fun main() {
    val t = readInt()
    for (i in 1 until t + 1) {
        val (n, k) = readInts()
        val a = readInts()
        val solution = Solution(n, k, a)
        solution.solve()
    }
}
